logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -392,7 +392,7 @@
     # We need to create theses dummy classes to know how many
     # events where actually generated by the pipeline and successfully send to the output.
     # Theses values are compared with what we store in the metric store.
-    class DummyOutput2 < DummyOutput; end
+    class DummyOutput2 < LogStash::Outputs::DummyOutput; end
```

Comment:
I would answer it depends.

When the first version of the registry was implemented, it stopped calling the `Plugin.namespace_lookup`, I think it was a regression in the way the legacy plugins where working. The code was kept in the project, if you look in the master branch you will that this method exist.

This PR reintroduce the call to that so it inforce the need to have the correctly defined plugin in the right namespace, which is how legacy style has been working for a lot of version. If the user want to define theirs own structure they will use the add method.

Most of the time in the code when we define a custom or one shot plugin we also make sure that the `Plugin.lookup` return the right class, except in two case in the current core specs.

This is why I've changed the name of the default mocks to make sure it does work and reflect how the namespace lookup actually works.